Libs/MRML/Widgets/qMRMLVolumeThresholdWidget.cxx

```diff
@@ -286,7 +277,26 @@ void qMRMLVolumeThresholdWidget::updateWidgetFromMRMLDisplayNode()
 
   const double min = d->VolumeDisplayNode->GetLowerThreshold();
   const double max = d->VolumeDisplayNode->GetUpperThreshold();
+
+  const double minRangeValue = std::min(min, d->MinRangeSpinBox->value());
+  const double maxRangeValue = std::max(max, d->MaxRangeSpinBox->value());
+  d->setRange(minRangeValue, maxRangeValue);
+
   d->VolumeThresholdRangeWidget->setValues(min, max );
 
   d->VolumeThresholdRangeWidget->blockSignals(wasBlocking);
 }
+
+// --------------------------------------------------------------------------
+void qMRMLVolumeThresholdWidget::setPopupWidgetVisible(bool visible)
+{
+  Q_D(qMRMLVolumeThresholdWidget);
+  if (visible)
+  {
+    d->PopupWidget->showPopup();
+  }
+  else
+  {
+    d->PopupWidget->hidePopup();
+  }
```

Comment:
```suggestion
  d->PopupWidget->showPopup(visible);
```

See https://github.com/commontk/CTK/blob/c4f1eeead234022e74de978126e3a54f648880e5/Libs/Widgets/ctkBasePopupWidget.h#L178

Reply:
I have tried the below, but haven't had success. What would be the syntax?
```c++
this->connect(this->RangeButton, SIGNAL(toggled(bool)),
                q, SLOT(this->PopupWidget->showPopup(bool)));
```
```c++
this->connect(this->RangeButton, SIGNAL(toggled(bool)),
                q, SLOT(q->PopupWidget->showPopup(bool)));
```

Reply:
Ok I've simplified `setPopupWidgetVisible` with the `showPopup(bool)` method rather than trying to get rid of `setPopupWidgetVisible` completely.

Reply:
The syntax is object1, SIGNAL(method1), object2, SLOT(method2) - something like this:

    this->connect(this->RangeButton, SIGNAL(toggled(bool)),
                this->PopupWidget, SLOT(showPopup(bool)));

Reply:
Makes sense. Thanks! I confirmed it worked and I have pushed an update with this change.